wagtail/admin/views/pages/delete.py

```diff
@@ -17,32 +17,59 @@ def delete(request, page_id):
         raise PermissionDenied
 
     with transaction.atomic():
-        for fn in hooks.get_hooks('before_delete_page'):
+        for fn in hooks.get_hooks("before_delete_page"):
             result = fn(request, page)
-            if hasattr(result, 'status_code'):
+            if hasattr(result, "status_code"):
                 return result
 
         next_url = get_valid_next_url_from_request(request)
+        parent_page_translations = page.get_parent().get_translations()
+        translations_to_delete = [
+            page
+            for page in page.get_translations()
+            if page.get_parent() in parent_page_translations
+        ]
+        translation_descendant_count = sum(
+            [p.get_descendants().count() for p in translations_to_delete]
+        )
```

Comment:
@KalobTaulien https://github.com/wagtail/wagtail/issues/7592#issuecomment-1034790403 states we will use a hook to gather this data and then the various actions will act upon it. 
The hook only runs when `WAGTAIL_I18N_ENABLED = True`, and for core we would implent this in simple_translation only when `WAGTAILSIMPLETRANSLATION_SYNC_PAGE_TREE = True`.

We discussed that https://github.com/wagtail/wagtail/pull/7984/files lays the foundation and you'be building on top of it.